### PR TITLE
Update crypto example

### PR DESCRIPTION
**Pull request by okbel**
### Description

This new example shows how to use the WebCryptoAPI on both Vercel Edge Functions and Edge Middleware.

### Demo URL
Preview URL

### Type of Change

- [ ] New Example
- [X] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

### CODE REVIEW

File: edge-functions/crypto/next-env.d.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /// <reference types="next" />
-/// <reference types="next/types/global" />
+/// <reference types="next/dist/styled-jsx-types/global" />
```
Comment (lfades) [inline edge-functions/crypto/next-env.d.ts]: Was this a manual change? Next.js should be removing this line entirely 🤔

File: edge-functions/crypto/package.json [hunk 2]
```diff
@@ -4,21 +4,21 @@
...
     "build": "next build",
-    "start": "next start",
-    "lint": "next lint"
+    "dev": "next dev",
+    "lint": "next lint",
+    "start": "next start"
   },
   "dependencies": {
-    "next": "^12.1.0",
+    "next": "12.1.7-canary.45",
```
Comment (lfades) [inline edge-functions/crypto/package.json:13]: Let's use `canary` here, as those versions change quite fast and we want every example to always be on the latest canary if there are new improvements on the way.

```suggestion
    "next": "canary",
```

File: edge-functions/crypto/pages/api/crypto.ts [hunk 3]
```diff
@@ -33,7 +41,12 @@ export async function middleware(req: NextRequest) {
...
       iv,
+      fromMiddleware,
     }),
     { headers: { 'Content-Type': 'application/json' } }
   )
 }
+
+export const config = {
+  runtime: 'experimental-edge',
+}
```
Comment (lfades) [inline edge-functions/crypto/pages/api/crypto.ts:52]: As in middleware.ts, makes more sense to have it in the top

File: edge-functions/crypto/README.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-name: Using Crypto in Edge Functions
+name: Using Crypto in Edge Functions and Middleware
```
Comment (lfades) [inline edge-functions/crypto/README.md:2]: This feels redundant because Middleware is an edge function.

**Review by lfades — CHANGES_REQUESTED** [review]
Added some comments

File: edge-functions/crypto/package.json (hunk 2 above)
Comment (okbel) [reply edge-functions/crypto/package.json:13]: It doesn't work with canary. It points to another version. Has to be a fixed version for now.

File: edge-functions/crypto/README.md (hunk 4 above)
Comment (okbel) [reply edge-functions/crypto/README.md:2]: It's not redundant. There's a difference between our Edge Middleware and our Experimental Edge functions (shown and distinguished in the docs)

File: edge-functions/crypto/next-env.d.ts (hunk 1 above)
Comment (balazsorban44) [reply edge-functions/crypto/next-env.d.ts]: See https://github.com/vercel/next.js/pull/37902

**Review by lfades — COMMENTED** [review]
@okbel the PR is now including more changes than expected for some reason 🤔

**Comment by okbel** [comment]
@lfades 
Ready to be checked! Pulled from another branch by mistake. Should be good to go now

File: edge-functions/crypto/next-env.d.ts (hunk 1 above)
Comment (lfades) [reply edge-functions/crypto/next-env.d.ts:2]: After https://github.com/vercel/next.js/pull/37964 it can be removed now.

**Review by lfades — APPROVED** [review]

**Pull request merged**